### PR TITLE
client-go/features: add Set method to the envvar impl

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/features/envvar.go [hunk 1]
```diff
@@ -94,6 +114,31 @@ func (f *envVarFeatureGates) Enabled(key Feature) bool {
...
+	newMapFromSetMethod := map[Feature]bool{}
+	for k, v := range f.getEnabledMapFromSetMethod() {
+		if k == featureName {
+			continue
+		}
+		newMapFromSetMethod[k] = v
+	}
+	newMapFromSetMethod[featureName] = featureValue
+	f.enabledViaSetMethod.Store(newMapFromSetMethod)
+	return nil
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/features/envvar.go:139]: this construct means that two parallel `.Set` calls will result in one winner and one loser.  If you lock just above this, you'll end up with two winners.

Comment (p0lyn0mial) [reply staging/src/k8s.io/client-go/features/envvar.go]: Yes, you are right. We can do better here. Thanks, I'm glad you spotted the issue.

**Pull request merged**